### ISSUE TITLE
hiprngDestroyGenerator sigabrt error

### ISSUE THREAD

**Issue by parallelo** [ROCmSoftwarePlatform/hcRNG#6]
Hi, 

I'm seeing a regression in hipCaffe due to hcRNG -- at application tear down, there's a hcRNG run-time error.  This started happening within the last few weeks.  

Here's the failure signature:  
```
...
I0628 02:21:28.550993 20034 solver.cpp:322] Optimization Done.
I0628 02:21:28.550995 20034 caffe.cpp:254] Optimization Done.
*** Aborted at 1498616488 (unix time) try "date -d @1498616488" if you are using GNU date ***
PC: @     0x7fe842d2b428 gsignal
*** SIGABRT (@0x4e42) received by PID 20034 (TID 0x7fe81d6f6700) from PID 20034; stack trace: ***
    @     0x7fe84810f390 (unknown)
    @     0x7fe842d2b428 gsignal
    @     0x7fe842d2d02a abort
    @     0x7fe842d6d7ea (unknown)
    @     0x7fe842d7637a (unknown)
    @     0x7fe842d7a53c cfree
    @     0x7fe846cb09b8 hiprngDestroyGenerator
    @           0x6b9b23 caffe::Caffe::~Caffe()
    @           0x6be81b boost::thread_specific_ptr<>::delete_data::operator()()
    @     0x7fe844655161 (unknown)
    @     0x7fe84465655e (unknown)
    @     0x7fe8481056ba start_thread
    @     0x7fe842dfd3dd clone
    @                0x0 (unknown)
```

git-bisect log results:  
```
/root/hcRNG# git bisect log      
git bisect start
# bad: [c2ada991344b84612c1b2d13101eb2d62fc2c9fc] Used pre allocated buffer in hiprngGenerate
git bisect bad c2ada991344b84612c1b2d13101eb2d62fc2c9fc
# good: [7bbe7384379375f17a2d0de368bf96d4e9bc2926] Merge remote-tracking branch 'origin/master'
git bisect good 7bbe7384379375f17a2d0de368bf96d4e9bc2926
# good: [a3dd9ac0c04407a0c9965fa5374658275ce7f42c] Remove the constrain - streamCount should be multiple of numberCount
git bisect good a3dd9ac0c04407a0c9965fa5374658275ce7f42c
# bad: [5ad7ee8a2538b5e6a5952bf44bbc3ea500a31250] Remove warnings
git bisect bad 5ad7ee8a2538b5e6a5952bf44bbc3ea500a31250
# bad: [4e235fc799017be44d384cd41210ee73e311e496] Avoid redundant memory allocation
git bisect bad 4e235fc799017be44d384cd41210ee73e311e496
# good: [f9f06feb42a3c47afd7b0f18831dcb44fecfa82f] Remove redundant test cases
git bisect good f9f06feb42a3c47afd7b0f18831dcb44fecfa82f
# first bad commit: [4e235fc799017be44d384cd41210ee73e311e496] Avoid redundant memory allocation

```

Potential problematic commit:  4e235fc799017be44d384cd41210ee73e311e496

When we revert to the following commit (immediately before the "culprit commit"), we no longer see the failure:  f9f06feb42a3c47afd7b0f18831dcb44fecfa82f

Let me know if you've got any ideas on the subject.  

Best, 

Jeff

**Comment by NEELMCW** [comment]
thanks Jeff

How can we reproduce this error ? To ensure that our changes make the fix

**Comment by Yuvaraj-mcw** [comment]
Fixed at commit [0dfd446](https://github.com/ROCmSoftwarePlatform/hcRNG/commit/0dfd44603fc6e9cb43be941fdcfd29084d67cfef)
